<?xml version="1.0" encoding="utf-8"?>

<robot
  name="wire_gripper">
 <!--########### adding the gripper ########### -->
  <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
    <robotNamespace>/</robotNamespace>
    <controlPeriod>0.01</controlPeriod> <!-- Control loop period in seconds -->
  </plugin>

  <link
    name="gripper_base_link">
    <inertial>
      <origin
        xyz="-0.019922 7.2898E-05 0.077395"
        rpy="0 0 0" />
      <mass
        value="1.3258" />
      <inertia
        ixx="0.0035252"
        ixy="2.2645E-06"
        ixz="-3.673E-05"
        iyy="0.0042085"
        iyz="-5.1365E-06"
        izz="0.0020359" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://add_post_pro_simulation/mesh/gripper_base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://add_post_pro_simulation/mesh/gripper_base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="gripper_Link">
    <inertial>
      <origin
        xyz="-0.0195 0.00014702 0.002743"
        rpy="0 0 0" />
      <mass
        value="0.027698" />
      <inertia
        ixx="9.9675E-06"
        ixy="7.5503E-08"
        ixz="-2.2624E-07"
        iyy="2.0524E-05"
        iyz="1.7057E-09"
        izz="2.7839E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://add_post_pro_simulation/mesh/gripper_Link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://add_post_pro_simulation/mesh/gripper_Link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_base_Joint"
    type="fixed">
      <origin
        xyz="0 0 0.004"
        rpy="0 0 3.14" />
    <parent
      link="J6" />
    <child      link="gripper_base_link" />
    <axis
      xyz="0 0 -1" />
  </joint>
  <joint
    name="gripper_Joint"
    type="prismatic">
    <origin
      xyz="0 0 0.02565"
      rpy="0 0 0.0" />
    <parent
      link="gripper_base_link" />
    <child
      link="gripper_Link" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="0"
      upper="0.021"
      effort="0"
      velocity="0" />
  </joint>

<!--########### adding transmissions ########### -->

  <transmission
    name="trans_gripper">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="gripper_Joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_gripper">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
<!--########### end of adding the gripper ########### -->
</robot>
